docs/source/tutorials/memory_optimizations.rst

```diff
@@ -0,0 +1,301 @@
+
+.. code-block:: bash
+
+  tune run <RECIPE> --config <CONFIG> \
+  optimizer=bitsandbytes.optim.PagedAdamW
+
+or by directly :ref:`modifying a config file<config_tutorial_label>`:
+
+.. code-block:: yaml
+
+  optimizer:
+    _component_: bitsandbytes.optim.PagedAdamW
+    lr: 2e-5
+
+.. _glossary_opt_in_bwd:
+
+Fusing Optimizer Step into Backward Pass
+----------------------------------------
+
+*What's going on here?*
+
+Stateful optimizers (e.g. optimizers which use momentum) are the default in modern deep learning due to their stable convergence properties.
+However, maintaining a state of gradient statistics comes at the cost of additional memory usage. An immediate alternative might be to
+turn to stateless optimizers such as `stochastic gradient descent <https://pytorch.org/docs/stable/generated/torch.optim.SGD.html>`_
+without momentum, which don't require any additional memory usage, but will likely result in worse convergence during training.
+
+Can we find a middle ground here? Let's consider a technique which enables the use of "stateful" optimizers such as `AdamW <https://pytorch.org/docs/stable/generated/torch.optim.AdamW.html>`_
+without the memory overhead of gradient statistics, and without sacrificing their desirable convergence properties.
+How is this possible, you might ask? By *completely removing the buffer of gradients* which are stored by the optimizer during its ``step()``.
+
+To understand how this works, we encourage you to read through the relevant PyTorch tutorial on this concept:
+`How to save memory by fusing the optimizer step into the backward pass <https://pytorch.org/tutorials/intermediate/optimizer_step_in_backward_tutorial.html>`_.
+
+
+*Sounds great! How do I use it?*
+
+.. todo ref full finetune recipe doc
+
+In torchtune, you can enable this feature using the ``optimizer_in_bwd`` flag, which is currently only supported in our
+single-device full finetune recipe. This feature works best when gradient memory is particularly large
+i.e. when using a stateful optimizer with a model with a lot of parameters, and when you don't need to use
+:ref:`gradient accumulation <glossary_grad_accm>`.
+
+.. _glossary_peft:
+
+Parameter Efficient Fine-Tuning (PEFT)
+--------------------------------------
+
+.. _glossary_lora:
+
+Low Rank Adaptation (LoRA)
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+
+*What's going on here?*
+
+You can read our tutorial on :ref:`finetuning Llama2 with LoRA<lora_finetune_label>` to understand how LoRA works, and how to use it.
+Simply stated, LoRA greatly reduces the number of trainable parameters, thus saving significant gradient and optimizer
+memory during training.
+
+*Sounds great! How do I use it?*
+
+You can finetune using any of our recipes with the ``lora_`` prefix, e.g. :ref:`lora_finetune_single_device<lora_finetune_recipe_label>`. These recipes utilize
+LoRA-enabled model builders, which we support for all our models, and also use the ``lora_`` prefix, e.g.
```

Comment:
I think we can make this paragraph smaller. Showing the tutorial + recipe should probably do the job.